tests/transforms/test_mitigate.py

```diff
@@ -326,3 +332,132 @@ def mitigated_circuit(w1, w2):
+    def test_diffability_autograd(self):
+        """Testing that the mitigated qnode can be differentiated and returns the correct gradient in autograd"""
+        qnode_noisy = qml.QNode(qfunc, dev_noisy)
+        qnode_ideal = qml.QNode(qfunc, dev_ideal)
+
+        scale_factors = [1., 2., 3.]
+
+        mitigated_qnode = mitigate_with_zne(scale_factors, fold_global, Richardson_extrapolate)(
+            qnode_noisy
+        )
+
+        theta = np.array([np.pi/4, np.pi/4], requires_grad=True)
+
+        res = mitigated_qnode(theta)
+        assert qml.math.allclose(res, out_ideal, atol=1e-2)
+        grad = qml.grad(mitigated_qnode)(theta)
+        grad_ideal = qml.grad(qnode_ideal)(theta)
+        grad_noisy = qml.grad(qnode_noisy)(theta)
+        assert qml.math.allclose(grad_ideal, grad_ideal_0)
+        assert qml.math.allclose(grad, grad_ideal, atol=1e-2)
+        assert qml.math.allclose(grad, grad_ideal, atol=1e-1)
+
+    @pytest.mark.jax
+    def test_diffability_jax(self):
+        """Testing that the mitigated qnode can be differentiated and returns the correct gradient in jax"""
+        import jax
+        import jax.numpy as jnp
+        qnode_noisy = qml.QNode(qfunc, dev_noisy)
+        qnode_ideal = qml.QNode(qfunc, dev_ideal)
+
+        scale_factors = jnp.array([1., 2., 3.])
+
+        mitigated_qnode = mitigate_with_zne(scale_factors, fold_global, Richardson_extrapolate)(
+            qnode_noisy
+        )
+
+        theta = jnp.array([np.pi/4, np.pi/4],)
+
+        res = mitigated_qnode(theta)
+        assert qml.math.allclose(res, out_ideal, atol=1e-2)
+        grad = jax.grad(mitigated_qnode)(theta)
+        grad_ideal = jax.grad(qnode_ideal)(theta)
+        grad_noisy = jax.grad(qnode_noisy)(theta)
+        assert qml.math.allclose(grad_ideal, grad_ideal_0)
+        assert qml.math.allclose(grad, grad_ideal, atol=1e-2)
+        assert qml.math.allclose(grad, grad_ideal, atol=1e-1)
+
+    @pytest.mark.xfail # getting a factor of 2 in the gradient, not sure why!
+    @pytest.mark.torch
+    def test_diffability_torch(self):
+        """Testing that the mitigated qnode can be differentiated and returns the correct gradient in torch"""
+        import torch
+
+        qnode_noisy = qml.QNode(qfunc, dev_noisy, interface="torch")
+        qnode_ideal = qml.QNode(qfunc, dev_ideal, interface="torch")
+
+        scale_factors = torch.tensor([1., 2., 3.]).type(torch.float64)
+
+        mitigated_qnode = mitigate_with_zne(scale_factors, fold_global, Richardson_extrapolate)(
+            qnode_noisy
+        )
+
+        theta = torch.tensor([np.pi/4, np.pi/4], requires_grad=True)
+
+        res = mitigated_qnode(theta)
+
+        assert qml.math.allclose(res, out_ideal, atol=1e-2) #True
+        res.backward()
+        grad = theta.grad
+        res_ideal = qnode_ideal(theta)
+        res_ideal.backward()
+        grad_ideal = theta.grad
+        assert qml.math.allclose(grad_ideal, grad_ideal_0) # False, wrong by a factor 2
```

Comment:
The gradient is wrong by an exact factor of 2 and I have no idea why that shouldbe the case. Everything else is identical and in the current implementation there is no exceptions for torch (or any other interface for that matter)